tools/webots.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import shutil
+import subprocess
+import sys
+import textwrap
+from pathlib import Path
+
+from termcolor import cprint
+
+import b
+
+
+def register(command):
+    command.help = "Build and run images for use with Webots"
+    subparsers = command.add_subparsers(help="sub-command help", dest="sub_command")
+
+    build_subcommand = subparsers.add_parser("build", help="Build the docker image for use with Webots")
+    build_subcommand.add_argument("roles", nargs="+", help="The roles to build for the image")
+
+    push_subcommand = subparsers.add_parser(
+        "push",
+        help=textwrap.dedent(
+            """
+            Push the simulation docker image to the TC registry. Run the aws CLI and login with the TC credentials before running this command.
+            See "Uploading Docker Images" in the Robocup "API Specifications for Virtual Soccer Competition" document for details.
+        """
+        ),
+    )
+
+    run_subcommand = subparsers.add_parser("run", help="Run the simulation docker image for use with Webots")
+    run_subcommand.add_argument("role", help="The role to run")
+
+
+def get_cmake_flags(roles_to_build):
+    roles_dir = os.path.join(b.project_dir, "roles")
+    roles_glob = os.path.join(roles_dir, "*.role")
+
+    available_roles = [Path(role_path).stem for role_path in glob.glob(roles_glob)]
+
+    # Ensure that all the roles requested are available
+    for role in roles_to_build:
+        if role not in available_roles:
+            print(f"role '{role}' not found")
+            sys.exit(1)
+
+    role_flags = [f"-DROLE_{role}=ON" for role in available_roles if role in roles_to_build] + [
+        f"-DROLE_{role}=OFF" for role in available_roles if role not in roles_to_build
+    ]
```

Comment:
I feel like there is a better way to do this, but cant think of it
```suggestion
    available_roles = [Path(role_path).stem for role_path in glob.glob(roles_glob)]
    role_flags = [f"-DROLE_{role}=ON" for role in available_roles if role in roles_to_build] + [
        f"-DROLE_{role}=OFF" for role in available_roles if role not in roles_to_build
    ]
```